src/app/public/modules/action-hub/action-hub.component.ts

```diff
@@ -17,30 +18,78 @@ let data: SkyActionHubData;
 export class SkyActionHubComponent {
   /**
    * Passes a SkyActionHubData object to build the action hub. The page loads until
-   * `title` has a value.
+   * `[data]` has a value.
    *
    * @param value
    */
   @Input()
   public set data(value: typeof data) {
-    this.needsAttention = value.needsAttention;
-    this.parentLink = value.parentLink;
-    this.recentLinks = value.recentLinks;
-    this.relatedLinks = value.relatedLinks;
-    this.title = value.title;
+    if (typeof value === 'undefined') {
+      this.loading = true;
+    } else {
+      this.loading = false;
+      this.needsAttention = value.needsAttention;
+      this.parentLink = value.parentLink;
+      this.recentLinks = SkyActionHubComponent.getRecentLinksSorted(
+        value.recentLinks
+      );
+      this.relatedLinks = SkyActionHubComponent.getRelatedLinksSorted(
+        value.relatedLinks
+      );
+      this.title = value.title;
+    }
   }
 
   public needsAttention: SkyActionHubNeedsAttention[];
 
   public parentLink: SkyPageLink;
 
-  public recentLinks: SkyPageLink[];
+  public recentLinks: SkyRecentLink[];
 
   public relatedLinks: SkyPageLink[];
 
   public title = '';
 
-  public get loading(): boolean {
-    return !this.title;
+  public loading: boolean = true;
+
+  private static getRecentLinksSorted(
```

Comment:
What would you think about either expanding this function to take in the amount of links and then passing in the `5` in order to future proof this for changes, or add a comment somehwhere in the function calling out that we made the decision just to do 5. 

I feel like the `slice` at the end could seem odd to someone coming into this code a year down the road and doesn't realize that this is where we are specifying this. 

Or - just another thought - should we just be doing the limit to 5 in the template where we only show 5 but the actual field has all of them? We actually just ran into this in `lookup` where we had code like this for a very specific use case but then we needed all the items for something new later and had to rewrite things to make that work. If we had the field have all of them now and then just limited the rendering to 5 then we wouldn't have to rework things in the future if we need all of them.